### PR TITLE
Replaced magic numbers and formulas with defines

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
#### Summary

```SUMMARY: None```

#### Purpose of change

Reduce number of magic numbers in the code and normalize defines in coordinate related formulas.

### CODE REVIEW

File: src/coordinates.h [hunk 1]
```diff
@@ -11,7 +11,7 @@
  * This is less obvious than one might think, for negative coordinates, so this
  * was created to give a definitive answer.
  *
- * 'absolute' is defined as the -actual- submap x,y * 12 + position in submap, and
+ * 'absolute' is defined as the -actual- submap x,y * SEEX/SEEY + position in submap, and
```
Comment (ZhilkinSerg) [inline src/coordinates.h]:
```suggestion
  * 'absolute' is defined as the -actual- submap x,y * SEEX + position in submap, and
```

File: src/coordinates.h [hunk 2]
```diff
@@ -23,10 +23,10 @@ struct real_coords {
     static const int subs_in_om_n = subs_in_om - 1;
 
     point abs_pos;     // 1 per tile, starting from tile 0,0 of submap 0,0 of overmap 0,0
-    point abs_sub;     // submap: 12 tiles.
+    point abs_sub;     // submap: SEEX tiles.
```
Comment (ZhilkinSerg) [inline src/coordinates.h]:
```suggestion
    point abs_sub;     // submap: 12 tiles.
```

File: src/coordinates.h [hunk 3]
```diff
@@ -23,10 +23,10 @@ struct real_coords {
...
     point abs_om;      // overmap: 360 submaps.
 
-    point sub_pos;     // coordinate (0-11) in submap / abs_pos constrained to % 12.
+    point sub_pos;     // coordinate (0-SEEX) in submap / abs_pos constrained to % SEEX.
```
Comment (ZhilkinSerg) [inline src/coordinates.h]:
```suggestion
    point sub_pos;     // coordinate (0-11) in submap / abs_pos constrained to % 12.
```

File: src/editmap.cpp [hunk 4]
```diff
@@ -2107,17 +2108,17 @@ int editmap::edit_mapgen()
...
             tc.fromabs( g->m.getabs( target.x, target.y ) );
         }
         target_list.clear();
-        for( int x = target.x - 11; x < target.x + 13; x++ ) {
-            for( int y = target.y - 11; y < target.y + 13; y++ ) {
-                if( x == target.x - 11 || x == target.x + 12 ||
-                    y == target.y - 11 || y == target.y + 12 ) {
+        for( int x = target.x - SEEX + 1; x < target.x + SEEX + 1; x++ ) {
+            for( int y = target.y - SEEY + 1; y < target.y + SEEY + 1; y++ ) {
+                if( x == target.x - SEEX + 1 || x == target.x + SEEX + 1 ||
```
Comment (ZhilkinSerg) [inline src/editmap.cpp]:
```suggestion
                if( x == target.x - SEEX + 1 || x == target.x + SEEX ||
```

File: src/editmap.cpp [hunk 5]
```diff
@@ -2107,17 +2108,17 @@ int editmap::edit_mapgen()
...
+                    y == target.y - SEEY + 1 || y == target.y + SEEY + 1 ) {
```
Comment (ZhilkinSerg) [inline src/editmap.cpp]:
```suggestion
                    y == target.y - SEEY + 1 || y == target.y + SEEY ) {
```

File: src/coordinates.h [hunk 6]
```diff
@@ -44,26 +44,26 @@ struct real_coords {
         abs_pos = point( absx, absy );
 
         if( absx < 0 ) {
-            abs_sub.x = ( absx - 11 ) / 12;
-            sub_pos.x = 11 - ( ( normx - 1 ) % 12 );
+            abs_sub.x = ( absx - SEEX + 1 ) / SEEX;
```
Comment (kevingranade) [inline src/coordinates.h]:
```suggestion
            abs_sub.x = ( absx - SEEX - 1 ) / SEEX;
```

Comment (kevingranade) [reply src/coordinates.h]: This preserves the previous value, or was the old value erroneous?

File: src/coordinates.h [hunk 7]
```diff
@@ -44,26 +44,26 @@ struct real_coords {
...
+            sub_pos.x = absx % SEEX;
             abs_om.x = abs_sub.x / subs_in_om;
             om_sub.x = abs_sub.x % subs_in_om;
         }
         om_pos.x = om_sub.x / 2;
 
         if( absy < 0 ) {
-            abs_sub.y = ( absy - 11 ) / 12;
-            sub_pos.y = 11 - ( ( normy - 1 ) % 12 );
+            abs_sub.y = ( absy - SEEY + 1 ) / SEEY;
```
Comment (kevingranade) [inline src/coordinates.h]:
```suggestion
            abs_sub.y = ( absy - SEEY - 1 ) / SEEY;
```

File: src/editmap.cpp [hunk 8]
```diff
@@ -1643,8 +1643,8 @@ int editmap::mapgen_preview( real_coords &tc, uilist &gmenu )
     // TODO: keep track of generated submaps to delete them properly and to avoid memory leaks
     tmpmap.generate( omt_pos.x * 2, omt_pos.y * 2, target.z, calendar::turn );
 
-    tripoint pofs = pos2screen( { target.x - 11, target.y - 11, target.z } );
-    catacurses::window w_preview = catacurses::newwin( 24, 24, pofs.y, pofs.x );
+    tripoint pofs = pos2screen( { target.x - SEEX + 1, target.y - SEEY + 1, target.z } );
```
Comment (kevingranade) [inline src/editmap.cpp]:
```suggestion
    tripoint pofs = pos2screen( { target.x - SEEX - 1, target.y - SEEY - 1, target.z } );
```

File: src/editmap.cpp [hunk 9]
```diff
@@ -2037,13 +2037,14 @@ int editmap::mapgen_retarget()
...
-            if( pinbounds( ptarget ) && inbounds( ptarget.x + 24, ptarget.y + 24, ptarget.z ) ) {
+            tripoint ptarget = tripoint( target.x + ( vec->x * SEEX * 2 ), target.y + ( vec->y * SEEY * 2 ),
+                                         target.z );
+            if( pinbounds( ptarget ) && inbounds( ptarget.x + SEEX * 2, ptarget.y + SEEY * 2, ptarget.z ) ) {
                 target = ptarget;
 
                 target_list.clear();
-                for( int x = target.x - 11; x < target.x + 13; x++ ) {
-                    for( int y = target.y - 11; y < target.y + 13; y++ ) {
+                for( int x = target.x - SEEX + 1; x < target.x + SEEX + 1; x++ ) {
```
Comment (kevingranade) [inline src/editmap.cpp]:
```suggestion
                for( int x = target.x - SEEX - 1; x < target.x + SEEX + 1; x++ ) {
```

File: src/editmap.cpp [hunk 10]
```diff
@@ -2037,13 +2037,14 @@ int editmap::mapgen_retarget()
...
+                    for( int y = target.y - SEEY + 1; y < target.y + SEEY + 1; y++ ) {
```
Comment (kevingranade) [inline src/editmap.cpp]:
```suggestion
                    for( int y = target.y - SEEY - 1; y < target.y + SEEY + 1; y++ ) {
```

File: src/editmap.cpp [hunk 11]
```diff
@@ -2107,17 +2108,17 @@ int editmap::edit_mapgen()
...
         if( target.x != om_ltarget.x || target.y != om_ltarget.y ) {
             target = om_ltarget;
             tc.fromabs( g->m.getabs( target.x, target.y ) );
         }
         target_list.clear();
-        for( int x = target.x - 11; x < target.x + 13; x++ ) {
-            for( int y = target.y - 11; y < target.y + 13; y++ ) {
-                if( x == target.x - 11 || x == target.x + 12 ||
-                    y == target.y - 11 || y == target.y + 12 ) {
+        for( int x = target.x - SEEX + 1; x < target.x + SEEX + 1; x++ ) {
```
Comment (kevingranade) [inline src/editmap.cpp]:
```suggestion
        for( int x = target.x - SEEX - 1; x < target.x + SEEX + 1; x++ ) {
```

File: src/editmap.cpp [hunk 12]
```diff
@@ -2107,17 +2108,17 @@ int editmap::edit_mapgen()
...
+            for( int y = target.y - SEEY + 1; y < target.y + SEEY + 1; y++ ) {
```
Comment (kevingranade) [inline src/editmap.cpp]:
```suggestion
            for( int y = target.y - SEEY - 1; y < target.y + SEEY + 1; y++ ) {
```

File: src/editmap.cpp [hunk 13]
```diff
@@ -2107,17 +2108,17 @@ int editmap::edit_mapgen()
...
+                if( x == target.x - SEEX + 1 || x == target.x + SEEX ||
```
Comment (kevingranade) [inline src/editmap.cpp]:
```suggestion
                if( x == target.x - SEEX - 1 || x == target.x + SEEX ||
```

File: src/editmap.cpp [hunk 14]
```diff
@@ -2107,17 +2108,17 @@ int editmap::edit_mapgen()
...
+                    y == target.y - SEEY + 1 || y == target.y + SEEY ) {
```
Comment (kevingranade) [inline src/editmap.cpp]:
```suggestion
                    y == target.y - SEEY - 1 || y == target.y + SEEY ) {
```

File: src/coordinates.h (hunk 6 above)
Comment (ZhilkinSerg) [reply src/coordinates.h]: This would change expression from  `( absx - 11 ) / 12` to `( absx - 13 ) / 12` which would be wrong.

Comment (kevingranade) [reply src/coordinates.h]: you are correct, stupid OOO and signs.

**Pull request merged**